### PR TITLE
Simplify icons structure

### PR DESCRIPTION
**Pull request by tfuxu**
This PR simplifies the way icons can be added to gtkcord4. It also removes all PNG icons and changes some of the icons to GNOME alternatives.

### CODE REVIEW

File: internal/icons/icons.go [hunk 1]
```diff
@@ -1,134 +1,24 @@
...
 
-//go:generate -command pack-icons go run ./pack-icons -s rounded -px 20
-//go:generate pack-icons svg/*.svg alternate_email add_reaction campaign forum chat mark_chat_unread mark_chat_read send
+//go:generate glib-compile-resources gtkcord4.gresource.xml
 
-//go:generate ./svg2png.sh
+//go:embed gtkcord4.gresource
+var Resources []byte
 
-//go:embed png/*.png
```
Comment (diamondburned) [inline internal/icons/icons.go:28]: Can this be in `init()`?

**Review by diamondburned — CHANGES_REQUESTED** [review]
This is an interesting PR! Is there a way for me to not have to touch the XML file for future icon additions? I also added a minor change.

File: internal/icons/icons.go (hunk 1 above)
Comment (diamondburned) [reply internal/icons/icons.go:28]: Sorry, this was supposed to refer to `func LoadResources() {`.

**Comment by tfuxu** [comment]
> This is an interesting PR! Is there a way for me to not have to touch the XML file for future icon additions? I also added a minor change.

I could probably make a script for this, but adding a new icon to GResource is just a matter of adding one extra line, like this one:
```xml
<file preprocess="xml-stripblanks" alias="avatar-default-symbolic.svg">scalable/actions/avatar-default-symbolic.svg</file>
```

File: internal/icons/icons.go (hunk 1 above)
Comment (tfuxu) [reply internal/icons/icons.go:28]: As `icons` module is now imported only once, I think it's better to put resource init code in a function and call it manually in main's `init`

Comment (diamondburned) [reply internal/icons/icons.go:28]: The codebase already makes liberal use of `init`, so I don't know if there's still a point in refraining to use it, especially when there's no reason to delay the call until `main.init`.

**Pull request merged**